drivers/accel/adxl355/iio_adxl355.c

```diff
@@ -0,0 +1,1032 @@
+enum adxl355_iio_accel_chans {
+	chan_x,
+	chan_y,
+	chan_z,
+};
+
+/******************************************************************************/
+/************************ Functions Declarations ******************************/
+/******************************************************************************/
+extern int adxl355_read_device_data(struct adxl355_dev *dev,
+				    uint8_t base_address, uint16_t size, uint8_t *read_data);
+extern int adxl355_write_device_data(struct adxl355_dev *dev,
+				     uint8_t base_address, uint16_t size, uint8_t *write_data);
+static int adxl355_iio_fill_3db_frequency_table(void *dev);
+static int adxl355_iio_find_2d_row(const int (*freq_tbl)[2], const int n,
+				   const int val1, const int val2);
+static int adxl355_iio_read_reg(struct adxl355_iio_dev *dev, uint32_t reg,
+				uint32_t *readval);
+static int adxl355_iio_write_reg(struct adxl355_iio_dev *dev, uint32_t reg,
+				 uint32_t writeval);
+static int adxl355_iio_read_raw(void *dev, char *buf, uint32_t len,
+				const struct iio_ch_info *channel, intptr_t priv);
+static int adxl355_iio_read_offset(void *dev, char *buf, uint32_t len,
+				   const struct iio_ch_info *channel, intptr_t priv);
+static int adxl355_iio_read_scale(void *dev, char *buf, uint32_t len,
+				  const struct iio_ch_info *channel, intptr_t priv);
+static int adxl355_iio_read_calibbias(void *dev, char *buf, uint32_t len,
+				      const struct iio_ch_info *channel, intptr_t priv);
+static int adxl355_iio_write_calibbias(void *dev, char *buf, uint32_t len,
+				       const struct iio_ch_info *channel, intptr_t priv);
+static int adxl355_iio_read_hpf(void *dev, char *buf, uint32_t len,
+				const struct iio_ch_info *channel, intptr_t priv);
+static int adxl355_iio_write_hpf(void *dev, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel, intptr_t priv);
+static int adxl355_iio_read_sampling_freq(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel, intptr_t priv);
+static int adxl355_iio_write_sampling_freq(void *dev, char *buf,
+		uint32_t len, const struct iio_ch_info *channel, intptr_t priv);
+static int adxl355_iio_read_hpf_available(void *dev, char *buf, uint32_t len,
+		const struct iio_ch_info *channel, intptr_t priv);
+static int adxl355_iio_read_samp_freq_avail(void *dev, char *buf,
+		uint32_t len, const struct iio_ch_info *channel, intptr_t priv);
+static int adxl355_iio_read_samples(void* dev, int* buff, uint32_t samples);
+static int adxl355_iio_update_channels(void* dev, uint32_t mask);
+
+/******************************************************************************/
+/************************ Variable Declarations ******************************/
+/******************************************************************************/
+static struct iio_attribute adxl355_iio_temp_attrs[] = {
+	{
+		.name = "offset",
+		.show = adxl355_iio_read_offset,
+		.store = NULL,
```

Comment:
I know there are a lot of examples in no-OS where unused pointers are initialized like this, but let's just not initialize them from now on, they get initialized to NULL by the compiler